### PR TITLE
Allow the link queue wrapper to wrap a generic type

### PR DESCRIPTION
**Pull request by RubenEschauzier**
PR to allow the wrapped link queue type to be generic (which defaults to the current implementation of the link queue wrapper). This will enable users to wrap other types of link queues, like a priority queue.

Sidenote: I thought the linter would fix the weird change log of ActorRdfJoin, but it didn't. If this is a problem let me know.

Credit to @joachimvh, as he suggested this implementation.

### CODE REVIEW

File: packages/bus-rdf-join/lib/ActorRdfJoin.ts [hunk 1]
```diff
@@ -279,7 +279,7 @@ export abstract class ActorRdfJoin
       parentPhysicalQueryPlanNode = action.context.get(KeysInitQuery.physicalQueryPlanNode);
       action.context = action.context.set(KeysInitQuery.physicalQueryPlanNode, action);
     }
-
```
Comment (rubensworks) [inline packages/bus-rdf-join/lib/ActorRdfJoin.ts:282]: Unneeded changes?

**Comment by joachimvh** [comment]
Had a look since I got tagged. Depending on how generic you want to go and what it is exactly that you're having issues with due to the fixed typing, you could also put this on the interface level. E.g., `export interface ILinkQueue<T extends ILink = ILink>` and then `export class LinkQueueWrapper<T extends ILink = ILink> implements ILinkQueue<T>`. That way you could also have generics in the response types of your functions.

File: packages/bus-rdf-join/lib/ActorRdfJoin.ts (hunk 1 above)
Comment (RubenEschauzier) [reply packages/bus-rdf-join/lib/ActorRdfJoin.ts:282]: Yep, I noticed my whole code base is a bit messed up. I'll open a new pull request which is cleaner hopefully

**Comment by RubenEschauzier** [comment]
That would be more flexible, but it seems to me that because the interface is called LinkQueue, you should expect to only queue links. You can already pass different types of links to the queue by extending the ILink interface.

**Comment by RubenEschauzier** [comment]
Closed in favor of the cleaner branch

**Pull request closed without merging**